version: '3.4'

services:
  autonomous_ros2:
    image: autonomous

    build:
      context: ..
      dockerfile: .devcontainer/ros2_ws.Dockerfile

    environment:
      - DISPLAY
      - NVIDIA_VISIBLE_DEVICES=0
      - NVIDIA_DRIVER_CAPABILITIES=all
      
      - ROS_DOMAIN_ID=47  # Its 47 for obvious reasons

    volumes:
      - ..:/ws

    network_mode: host
    ipc: host
    stdin_open: true
    tty: true

    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              device_ids: []
              capabilities: [gpu]


  autonomous_zed:
    image: zed_sdk

    privileged: true

    runtime: nvidia

    
    build:
      context: ..
      dockerfile: .devcontainer/zed.Dockerfile

    environment:
      - DISPLAY
      - NVIDIA_VISIBLE_DEVICES=0
      - NVIDIA_DRIVER_CAPABILITIES=all
      - ROS_DOMAIN_ID=47


    volumes:
      - ../src/perception/:/ws/src/perception/
      - ../src/moa/moa_description:/ws/src/moa/moa_description
      - ../src/moa/moa_msgs:/ws/src/moa/moa_msgs
      - ../ros_entrypoint.sh:/ws/ros_entrypoint.sh
      - /tmp/.X11-unix:/tmp/.X11-unix
      - /dev:/dev
    
    command: bash -c ". /ws/ros_entrypoint.sh && ros2 launch zed_wrapper zed_camera.launch.py camera_model:='zed2i'"

    network_mode: host
    ipc: host
    stdin_open: true
    tty: true

    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              device_ids: []
              capabilities: [gpu]
